database/schema/V1.5__schema.sql

```diff
@@ -1,5 +1,13 @@
 -- create user
-create user app_user;
+do
+$$
+BEGIN
+	IF NOT EXISTS (SELECT * FROM pg_user WHERE usename = 'app_user') AND NOT EXISTS (SELECT * FROM pg_roles WHERE rolname = 'app_user') THEN
+		CREATE USER app_user;
+	END IF;
+END
+$$;
+
```

Comment:
```suggestion
DO
$$
BEGIN
    IF NOT EXISTS (SELECT 1 FROM pg_roles WHERE rolname = 'app_user') THEN
       CREATE USER app_user;
    END IF;
END
$$;

```

I think you can just check the pg_roles, cos I believe if you're in pg_user you're in pg_roles cos pg_roles stores all the users and roles. cos if I'm right Postgres, every user is also a role.

Reply:
See I thought that too, but when I had it set up this way it wouldn't work. I ran the queries locally and it seemed like the roles are not stored in the users table like I thought. There does seem to be some dependency/rule which checks to see if a role and user have the same name, and then disallows it if they do

Reply:
Thinking about it now I think you might be right. I may have checked it the other way round, thinking that the users table keeps all the roles as well. Let me check it locally again and I'll amend if needed

Reply:
You were correct. I've made the change 👍